gpii/node_modules/gpii-oauth2/gpii-oauth2-authz-server/public/src/components/selectionTree/js/SelectionTree.js

```diff
@@ -0,0 +1,442 @@
+/*!
+GPII OAuth2 server
+
+Copyright 2014-2015 OCAD University
+
+Licensed under the New BSD license. You may not use this file except in
+compliance with this License.
+
+You may obtain a copy of the License at
+https://github.com/gpii/universal/LICENSE.txt
+*/
+
+// Declare dependencies
+/* global fluid, jQuery */
+
+var gpii = gpii || {};
+
+(function ($) {
+    "use strict";
+
+    fluid.registerNamespace("gpii.oauth2.selectionTree");
+
+    gpii.oauth2.selectionTree.preventDefault = function (e) {
+        e.preventDefault();
+    };
+
+    fluid.defaults("gpii.oauth2.selectionTree", {
+        gradeNames: ["fluid.viewRelayComponent", "autoInit"],
+        selectors: {
+            tree: ".gpiic-oauth2-selectionTree-tree",
+            leaf: ".gpiic-oauth2-selectionTree-leaf",
+            branch: ".gpiic-oauth2-selectionTree-branch",
+            branchToggle: ".gpiic-oauth2-selectionTree-branchToggle",
+            prefereces: "label"
+        },
+        domMap: {}, // must supply a mapping with a key that is an ELPath and a value of the corresponding selector
+        styles: {
+            collapsed: "gpii-icon-plus-small",
+            expanded: "gpii-icon-minus-small",
+            select: "gpii-oauth2-focus"
+        },
+        model: {},
+        requestedPrefs: {},
+        resources: {
+            template: {
+                href: "../html/SelectionTreeTemplate.html",
+                forceCache: true
+            }
+        },
+        invokers: {
+            refresh: {
+                funcName: "gpii.oauth2.selectionTree.refresh",
+                args: ["{that}"]
+            },
+            toModel: {
+                funcName: "gpii.oauth2.selectionTree.toModel",
+                args: ["{arguments}.0", "{that}.options.requestedPrefs"]
+            },
+            toServerModel: {
+                funcName:"gpii.oauth2.selectionTree.toServerModel",
+                args: ["{that}.model"]
+            },
+            updateModelFromServer: {
```

Comment:
Add a TODO here that the whole "serverModel" logic in this component needs to be offloaded from this implementation, as we discussed in December. We can leave this task for when we have more time and are refactoring this whole UI to be payload- rather than markup-driven